proposals.md

```diff
@@ -62,55 +21,212 @@ On the other side of the equation are the community members who are reviewing th
 
+## Proposed templates for Design and Build Phases
 
+The templates for this must include the following structure:
 
+### Design Phase Template
 
+1. What is being proposed?
+2. Why is this being proposed?
+3. What is in scope for this proposal?
+4. What is deliberately *not* in scope?
+5. What alternatives have been considered?
 
 
+### Build Phase Template
 
+1. How will this work, technically?
+  * Design documents
+  * System diagrams
+  * Code examples
+2. How will success be measured?
+  * Performance targets
+  * Compabitility
+3. Checklist of things that will must be done before this proposal is complete
+  * Code changes
+  * Tests added (e2e, unit)
+  * SDK changes (if needed)
+  * Documentation
 
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+## Feature lifecycle
+
+TBD
+
+### Proposed New API lifecycle
+
+An API is one form of feature that may be proposed by the contributor. Any new APIs can be split into two categories:
+- new API for an existing building block e.g.: adding query state API to state management building block 
+- new API in a new building block e.g.: configuration building block and adding APIs for that
+
+The scope of this definition is limited to the `dapr/dapr` and `dapr/components-contrib` repos. 
+`dapr/docs`, `dapr/quickstarts` repositories may be touched upon referring to the documentation/quickstart updates needed related to the new API definition.
+SDK documentation may provide guidance on requirements to enhance the SDK. The SDKs will be touched upon only to mention the requirement of having implementation for the new APIs.
+Similarly addition of `dapr/cli` commands may be mentioned, but the guidance for that is out of scope for this proposal.
+
+Additionally the scope is limited to definition of a new API only. Enhancement to existing stable APIs shall be done seperately.
+
+#### Proposed New API lifecycle stages
+
+For new APIs, it is proposed to set three stages for the lifecycle of the API:
+- Alpha
+  - API is not production ready yet and might contain bugs
+  - Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases
+- Beta
+  - API is not production ready yet
+  - Multiple components implement the API and API contract is mostly finalized
+  - Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases
+- Stable
+  - API is ready for production usage
+  - Performance numbers are published for the API
```

Comment:
I think we need a clear definition of what a Feature lifecycle is going to look like ... If a feature is an enhancement to an existing API, what would be lifecycle stage of the existing API ? Will it move from Stable to Beta ... which i think should not happen at all ... So probably Enhancements will have all the requirements that the Stable level APIs have.

Reply:
I will merge this and also integrate some other parts of your doc